libs/sdk/skyux-eslint/src/rules/template/prefer-label-text.spec.ts

```diff
@@ -90,6 +90,128 @@ ruleTester.run(RULE_NAME, rule, {
+        <sky-input-box>
+        ~~~~~~~~~~~~~~~~
+          <label>My label</label>
+          ~~~~~~~~~~~~~~~~~~~~~~~
+          <input class="sky-form-control" type="text" />
+          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        </sky-input-box>
+        ~~~~~~~~~~~~~~~~
+      `,
+      annotatedOutput: `
+        <sky-input-box labelText="My label">
+        ~
+          ~
+          ~
+          <input  type="text" />
+          ~
+        </sky-input-box>
+        ~
+      `,
+
+      messageId,
+      data: {
+        selector: 'sky-input-box',
+        labelInputName: 'labelText',
+        labelSelector: 'label',
+      },
+    }),
+    convertAnnotatedSourceToFailureCase({
+      description:
+        'should remove `sky-form-control` class from inputs with multiple classes',
+      annotatedSource: `
+        <sky-input-box>
+        ~~~~~~~~~~~~~~~~
+          <label>My label</label>
+          ~~~~~~~~~~~~~~~~~~~~~~~
+          <textarea class="my-input sky-form-control foobar"></textarea>
+          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        </sky-input-box>
+        ~~~~~~~~~~~~~~~~
+      `,
+      annotatedOutput: `
+        <sky-input-box labelText="My label">
+        ~
+          ~
+          ~
+          <textarea class="my-input foobar"></textarea>
+          ~
+        </sky-input-box>
+        ~
+      `,
+
+      messageId,
+      data: {
+        selector: 'sky-input-box',
+        labelInputName: 'labelText',
+        labelSelector: 'label',
+      },
+    }),
+    convertAnnotatedSourceToFailureCase({
+      description:
+        'should remove `sky-form-control` class from inputs if first in list of classes',
+      annotatedSource: `
+        <sky-input-box>
+        ~~~~~~~~~~~~~~~~
+          <label>My label</label>
+          ~~~~~~~~~~~~~~~~~~~~~~~
+          <select class="sky-form-control foo"></select>
+          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        </sky-input-box>
+        ~~~~~~~~~~~~~~~~
+      `,
+      annotatedOutput: `
+        <sky-input-box labelText="My label">
+        ~
+          ~
+          ~
+          <select class="foo"></select>
+          ~
+        </sky-input-box>
+        ~
+      `,
+
+      messageId,
+      data: {
+        selector: 'sky-input-box',
+        labelInputName: 'labelText',
+        labelSelector: 'label',
+      },
+    }),
+    convertAnnotatedSourceToFailureCase({
+      description: 'should remove skyId from sky-input-box inputs',
+      annotatedSource: `
+        <sky-input-box>
+        ~~~~~~~~~~~~~~~~
+          <label>My label</label>
+          ~~~~~~~~~~~~~~~~~~~~~~~
+          <input #myInput="skyId" skyId type="text" />
+          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        </sky-input-box>
+        ~~~~~~~~~~~~~~~~
+      `,
+      annotatedOutput: `
+        <sky-input-box labelText="My label">
+        ~
+          ~
+          ~
+          <input   type="text" />
```

Comment:
Is there a way to remove the extra spaces here?

Reply:
Unfortunately, not easily. The AST doesn't "know" about whitespace, and it would be difficult to determine if the whitespace is inline, or just a newline without doing some parsing after the recorder applies its fixes (see below). Our best bet is to rely on Prettier, or manual adjustments by the user.
```
<input
  type="text"
/>
```
vs
```
<input type="text" />
```